src/main/kotlin/controller/ProjectController.kt

```diff
@@ -2,28 +2,82 @@ package edu.erittenhouse.gitlabtimetracker.controller
 
 import edu.erittenhouse.gitlabtimetracker.controller.result.ProjectFetchResult
 import edu.erittenhouse.gitlabtimetracker.gitlab.GitlabAPI
+import edu.erittenhouse.gitlabtimetracker.io.SettingsManager
 import edu.erittenhouse.gitlabtimetracker.model.Project
+import edu.erittenhouse.gitlabtimetracker.ui.util.suspension.SuspendingController
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.javafx.JavaFx
 import kotlinx.coroutines.withContext
-import tornadofx.Controller
-import tornadofx.asObservable
+import tornadofx.*
 
-class ProjectController : Controller() {
+class ProjectController : SuspendingController() {
     private val gitlabAPI by inject<GitlabAPI>()
     private val credentialController by inject<CredentialController>()
+    private val settings = SettingsManager(find<StorageConfig>().fileLocation)
     val projects = mutableListOf<Project>().asObservable()
 
+    private var pinnedProjects = listOf<Project>()
+    private var unpinnedProjects = listOf<Project>()
+
     /**
-     * Fetches user's projects from GitLab and
+     * Fetches user's projects from GitLab and adds them to the projects list
      */
     suspend fun fetchProjects(): ProjectFetchResult {
         val credentials = credentialController.credentials ?: return ProjectFetchResult.NoCredentials
-        val fullProjectList = gitlabAPI.project.listUserMemberProjects(credentials).map { Project.fromGitlabDto(it) }
+        val pinnedProjectIDs = settings.getPinnedProjects()
+        val fullProjectList = gitlabAPI.project.listUserMemberProjects(credentials).map { Project.fromGitlabDto(it, isPinned = it.id in pinnedProjectIDs) }
+        val (localPinnedProjects, localUnpinnedProjects) = fullProjectList.partition { it.pinned }
```

Comment:
These two lines could be merged. Also, we might want to turn these two operations into a sequence if possible.